src/sqlfluff/core/parser/lexer.py

```diff
@@ -21,16 +21,50 @@
 lexer_logger = logging.getLogger("sqlfluff.lexer")
 
 
-class LexMatch(namedtuple("LexMatch", ["new_string", "new_pos", "segments"])):
+@dataclass
```

Comment:
Are these objects mutable? If they are immutable, did you think about using `typing.NamedTuple`?

Reply:
Good idea :+1: